xarray/tests/test_dataarray.py

```diff
@@ -3305,6 +3305,45 @@ def test_rank(self):
         y = DataArray([0.75, 0.25, np.nan, 0.5, 1.0], dims=('z',))
         assert_equal(y.rank('z', pct=True), y)
 
+    def test_corr(self):
+        # self: Load demo data and trim it's size
+        ds = xr.tutorial.load_dataset('air_temperature')
+        air = ds.air[:18, ...]
+        # other: select missaligned data, and smooth it to dampen the correlation with self.
+        air_smooth = ds.air[2:20, ...].rolling(time=3, center=True).mean(dim='time')  # .
+        # A handy function to select an example grid
+
+        def select_pts(da):
+            return da.sel(lat=45, lon=250)
+
+        # Test #1: Misaligned 1-D dataarrays with missing values
+        ts1 = select_pts(air.copy())
+        ts2 = select_pts(air_smooth.copy())
+
+        def pd_corr(ts1, ts2):
+            """Ensure the ts are aligned and missing values ignored"""
+            # ts1,ts2 = xr.align(ts1,ts2)
+            valid_values = ts1.notnull() & ts2.notnull()
+
+            ts1 = ts1.where(valid_values, drop=True)
+            ts2 = ts2.where(valid_values, drop=True)
+
+            return ts1.to_series().corr(ts2.to_series())
+
+        expected = pd_corr(ts1, ts2)
+        actual = ts1.corr(ts2)
+        np.allclose(expected, actual)
```

Comment:
You can use `assert_allclose` from `xarray.testing`. I'm not sure whether that is asserting or returning a bool

Reply:
@max-sixty `assert_allclose` gives `AssertionError`, hence used `np.allclose` - it returns a bool.

Reply:
You need to use an actual assertion here. Otherwise this isn't testing anything -- `np.allclose()` could fail and we wouldn't know.